### PR TITLE
Feature/lra 14 put product saga

### PR DESCRIPTION
**Pull request by Shalin21**
feat(Update product saga): add update saga
feat(Put Product Saga): Add tests for product actions and reducer
refactor(Put Product Saga): Refactor of types for product
refactor(Put Product Saga): Refactor of productSaga exports
refactor(Routing): refactor router to be able to use history
in sagas

### CODE REVIEW

File: src/__tests__/actions/productActions.test.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import * as actions from '../../store/actions/productActions';
+import { IProduct } from '../../types/product';
+import { productActionTypes } from '../../store/actionTypes/productActionTypes';
+
+const tempProduct: IProduct = {
```
Comment (mkozachok) [inline src/__tests__/actions/productActions.test.ts]: This mock is using in two places, you can move to file or export from test reducer filer to this one.

File: src/components/forms/UpdateProductForm/UpdateProductForm.tsx [hunk 2]
```diff
@@ -1,55 +1,59 @@
...
-  );
-  const initialValues: IProduct = product;
-
   useEffect(() => {
-    dispatch(getProductByIdStart(props.product_id));
+      dispatch(getProductByIdStart(props.product_id));
   }, [dispatch, props.product_id]);
+  
+  useEffect(() => {
+    return () => console.log('end')
```
Comment (mkozachok) [inline src/components/forms/UpdateProductForm/UpdateProductForm.tsx]: Please remove this hook, my bad.

**Review by mkozachok — APPROVED** [review]

File: src/components/forms/UpdateProductForm/UpdateProductForm.tsx (hunk 2 above)
Comment (Shalin21) [reply src/components/forms/UpdateProductForm/UpdateProductForm.tsx]: Fixed

File: src/__tests__/actions/productActions.test.ts (hunk 1 above)
Comment (Shalin21) [reply src/__tests__/actions/productActions.test.ts]: Done

**Pull request merged**